src/mlpack/methods/hoeffding_trees/hoeffding_tree_impl.hpp

```diff
@@ -224,6 +224,117 @@ HoeffdingTree<FitnessFunction, NumericSplitType, CategoricalSplitType>::
   }
 }
 
+// Move constructor.
+template<typename FitnessFunction,
+         template<typename> class NumericSplitType,
+         template<typename> class CategoricalSplitType>
+HoeffdingTree<FitnessFunction, NumericSplitType, CategoricalSplitType>::
+    HoeffdingTree(HoeffdingTree&& other) :
+    numericSplits(std::move(other.numericSplits)),
+    categoricalSplits(std::move(other.categoricalSplits)),
+    dimensionMappings(other.dimensionMappings),
+    ownsMappings(true),
+    numSamples(other.numSamples),
+    numClasses(other.numClasses),
+    maxSamples(other.maxSamples),
+    checkInterval(other.checkInterval),
+    minSamples(other.minSamples),
+    datasetInfo(other.datasetInfo),
+    ownsInfo(true),
+    successProbability(other.successProbability),
+    splitDimension(other.splitDimension),
+    majorityClass(other.majorityClass),
+    majorityProbability(other.majorityProbability),
+    categoricalSplit(std::move(other.categoricalSplit)),
+    numericSplit(std::move(other.numericSplit))
+{
+  // Remove pointers.
+  other.dimensionMappings = nullptr;
+  other.datasetInfo = nullptr;
```

Comment:
Should we set the other members of `other` back to their defaults?  I'm not sure whether it's really necessary.

Reply:
I think because only two of them are pointers, then we have to set the other's pointers to null. For the other, there are variables that are objects (E.g: `categoricalSplits` is an `std::vector`), they have been converted to rvalue reference by `std::move`, other variables are primary type so they don't need to be set to null.

Reply:
Well I think for the primary type variables, I should have reset it to default value! :)